lib/chef/knife/s3_source.rb

```diff
@@ -0,0 +1,40 @@
+require 'fog'
+
+class Chef
+  class Knife
+    class S3Source
+      attr_accessor :url
+
+      def self.fetch(url)
+        source = Chef::Knife::S3Source.new
+        source.url = url
+        source.body
+      end
+
+      def body
+        bucket_obj.files.get(path).body
+      end
+
+      private
+
+      def bucket_obj
+        @bucket_obj ||= fog.directories.get(bucket)
+      end
+
+      def bucket
+        URI(@url).host
```

Comment:
Hi,

I'm not sure whether I should tell you that, please read it yourself <http://ntaprophengi.spineconsults.com/oijpuds>

Sincerely, hebgb@blueyonder.co.uk